### PR TITLE
Add example for icu_locale

### PR DESCRIPTION
**Pull request by zbraniecki**
https://github.com/unicode-org/icu4x/issues/187 for Locale.

### CODE REVIEW

**Comment by zbraniecki** [comment]
Without icu_locale: 321464 bytes
With: 327856 bytes

The crate adds 6392 bytes.

File: components/locale/examples/langid.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    // 1. Parse input strings into LanguageIdentifiers
+    let langids = LANGIDS
+        .iter()
+        .map(|s| s.parse())
+        .collect::<Result<Vec<LanguageIdentifier>, _>>()
+        .expect("Failed to parse Language Identifier");
+
+    {
+        // 2. Filter list of identifiers that match a given Language Identifier.
+        let reference: LanguageIdentifier = "en-US"
```
Comment (sffc) [inline components/locale/examples/langid.rs:27]: Use your macro here instead?  There should no longer be any reason to hard-code a `"...".parse().expect("...")` pattern for LanguageIdentifier.

File: components/locale/examples/langid.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            .parse()
+            .expect("Failed to parse Language Identifier");
+
+        let _matching = langids.iter().filter(|langid| *langid == &reference);
+    }
+
+    {
+        // 3. Filter list of identifiers that match a given language subtag.
+        let reference: subtags::Language = "en".parse().expect("Failed to parse Language");
```
Comment (sffc) [inline components/locale/examples/langid.rs:36]: Likewise: use `language!` (or `icu_language!`)

**Review by sffc — COMMENTED** [review]
This example seems a little synthetic.  Can you make it "do" something?  For example, given a list of locale strings, find all the ones with language subtag "en", and then canonicalize that filtered list.  You end up with a program that calculates all forms of English your app supports.

File: components/locale/examples/langid.rs (hunk 1 above)
Comment (zbraniecki) [reply components/locale/examples/langid.rs:27]: Once we merge the macros! :)

**Comment by zbraniecki** [comment]
> This example seems a little synthetic. Can you make it "do" something? For example, given a list of locale strings, find all the ones with language subtag "en", and then canonicalize that filtered list. You end up with a program that calculates all forms of English your app supports.

Good idea! I'll improve the example tmrw.

**Comment by zbraniecki** [comment]
Refactored the example according to @sffc's suggestion.

File: components/locale/examples/filter_langids.rs [hunk 3]
```diff
@@ -0,0 +1,54 @@
+// A sample application which takes a comma separated list of language identifiers,
+// filters out identifeirs with language subtags different than `en` and serializes
```
Comment (sffc) [inline components/locale/examples/filter_langids.rs:2]: Optional: s/identifeirs/identifiers

**Review by sffc — APPROVED** [review]

File: components/locale/examples/langid.rs (hunk 1 above)
Comment (sffc) [reply components/locale/examples/langid.rs]: I think it would be good to use the macros in this example.  Hopefully the macros will be checked in soon.

**Review by sffc — APPROVED** [review]

File: components/locale/examples/filter_langids.rs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+fn filter_input(input: &str) -> String {
+    // 1. Parse the input string into a list of language identifiers.
+    let langids: Vec<LanguageIdentifier> = input
+        .split(',')
+        .filter_map(|s| s.trim().parse().ok())
+        .collect();
+
+    // 2. Filter for LanguageIdentifiers with Language subtag `en`.
+    let en_lang: subtags::Language = "en".parse().expect("Failed to parse language subtag.");
```
Comment (filmil) [inline components/locale/examples/filter_langids.rs:19]: Is this a good place to demo the macros?

**Review by filmil — APPROVED** [review]

File: components/locale/examples/filter_langids.rs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    // 3. Serialize the output.
+    let en_strs: Vec<String> = en_langids.map(|langid| langid.to_string()).collect();
+
+    en_strs.join(", ")
+}
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+
+    let (_input, _output) = if let Some(input) = args.get(1) {
```
Comment (echeran) [inline components/locale/examples/filter_langids.rs:34]: Optional/minor Rust question: Would it have been possible for this to be written as:

```rust
    let _input = if let Some(input) = args.get(1) {
        input.as_str()
    } else {
        DEFAULT_INPUT
    };
    let _ouptut = filter_input(&input);
...
    println!("...", &_input, &_output);
```

I forgot if `println!` complains on `&String` or not.

**Review by echeran — DISMISSED** [review]
LGTM.  Had a minor Rust question, but it is unrelated to the example code.

File: components/locale/examples/filter_langids.rs (hunk 4 above)
Comment (zbraniecki) [reply components/locale/examples/filter_langids.rs:19]: once we land them, I'll switch to them in examples :)

File: components/locale/examples/filter_langids.rs (hunk 5 above)
Comment (zbraniecki) [reply components/locale/examples/filter_langids.rs]: I like it! Applied! thank you!

**Review by nciric — DISMISSED** [review]
Optional: Note somewhere that this is not the best way to do locale matching - use LanguageMatcher (TBD) instead.

**Pull request merged**